Feature/fix filled using last cell dynamic cell

andykkt opened the pull request:
Fixed an issue with dequeueReusableCell(withIdentifier:, for: indexPath) to get dynamic cell

andykkt committed:
Merge pull request #3 from BareFeetWare/develop

Develop

Commit:
Fixed an issue with dequeueReusableCell(withIdentifier:, for: indexPath) to get dynamic cell

barefeettom commented on BFWControls/Modules/Table/Controller/StaticTableViewController.swift:
```diff
@@ -85,6 +88,29 @@ class StaticTableViewController: UITableViewController {
                          section: (indexPath as NSIndexPath).section)
     }
+	
+	fileprivate func refreshTableView() {
```
Let's call this `updateCellHeights` or `refreshCellHeights`

andykkt replied:
We already have updateCellHeights function from UITableViewController extension in MOA, so I will change it to refreshCellHeights

barefeettom commented on BFWControls/Modules/Table/Controller/StaticTableViewController.swift:
```diff
@@ -85,6 +88,29 @@ class StaticTableViewController: UITableViewController {
+			? UITableViewAutomaticDimension
+			: super.tableView(tableView, heightForRowAt: superIndexPath(for: lastCellIndexPath))
+		// Use default height when last cell is not on the screen.
+		guard tableView.indexPathsForVisibleRows?.contains(lastCellIndexPath) == true
```
Delete the redundant `== true`

andykkt replied:
Fixed in latest commit

barefeettom commented on BFWControls/Modules/Table/Controller/StaticTableViewController.swift:
```diff
@@ -85,6 +88,29 @@ class StaticTableViewController: UITableViewController {
+		// Get height of empty spaces to fill
+		let availableHeight = tableView.frame.size.height - previousRowFrame.maxY
+		if let lastCell = tableView.cellForRow(at: lastCellIndexPath) {
+			if availableHeight.rounded() >= lastCell.frame.height.rounded() {
```
Why `rounded()`?

andykkt replied:
Some time it is not matching even though it sets same value. Only difference is on decimal points so I rounded the value to figure out whether it is greater than or equal to.

barefeettom commented on BFWControls/Modules/Table/Controller/StaticTableViewController.swift:
```diff
@@ -111,20 +137,40 @@ class StaticTableViewController: UITableViewController {
     
     // MARK: - UITableViewDelegate
+	override func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+		if filledUsingLastCell && isDynamicLastCell && (self.dynamicLastCellHeight == nil) {
```
Can we delete `self.` here? Can we delete the `(` and `)`?

andykkt replied:
Removed from latest commit

barefeettom commented on BFWControls/Modules/Table/Controller/StaticTableViewController.swift:
```diff
@@ -111,20 +137,40 @@ class StaticTableViewController: UITableViewController {
+					height = dynamicLastCellHeight
+				}
+			} else {
+				let previousRowFrame = tableView.rectForRow(at: IndexPath(row: indexPath.row - 1, section: indexPath.section))
```
Will this crash if there is only one cell?

andykkt replied:
It will not crash just previousRowFrame will not have any value, in that case availableHeight will always be smaller than lastCellIntrinsicHeight which will never be applied

Commit:
Changed function name refreshTableView to refreshCellHeights
Fixed problem with updating last cell when it is off the screen
Fixed syntax issues
